execution_engine/src/runtime_context/mod.rs

```diff
@@ -945,6 +953,49 @@ where
         Ok(())
     }
 
+    /// Emits message and writes message summary to global state with a measurement.
+    pub(crate) fn metered_emit_message<V>(
+        &mut self,
+        topic_key: Key,
+        topic_value: V,
+        message_key: Key,
+        message_value: V,
+        message: Message,
+    ) -> Result<(), Error>
+    where
+        V: Into<StoredValue>,
```

Comment:
I find this function signature a bit odd, the function immediately validates that the input is a `MessageTopic` and `MessageValue`, but we allow any `Into<StoredValue>` as an argument
What if we just forced the caller to give as a pair of `MessageTopicSummary` and `MessageChecksum` and then wrap them with `StoredValue` inside the funciton, which can potentially get rid of the need for validation?